x/leverage/keeper/token.go

```diff
@@ -20,7 +18,7 @@ func (k Keeper) SetTokenSettings(ctx sdk.Context, token types.Token) error {
 
 	bz, err := k.cdc.Marshal(&token)
 	if err != nil {
-		panic(fmt.Errorf("failed to encode token settings: %w", err))
```

Comment:
Why remove the panic?

Reply:
Since the function has an error return, we can just pass the error out that way. (It it's in `EndBlock` after a gov proposal the error will turn into a panic anyway, but will have the original error message)